app/controllers/api_keys_controller.rb

```diff
@@ -1,33 +1,24 @@
-class ApiKeysController < ApplicationController
-  before_action :set_organisation_id
-  before_action :set_api_key_id, only: %i[update revoke delete]
+class ApiKeysController < AuthenticatedController
+  before_action :set_api_key, only: %i[update revoke delete]
 
   def index
-    @api_keys = ApiKey.all
+    @api_keys = ApiKey.where(organisation_id: organisation_id)
   end
 
   def new; end
 
-  def show
-    if params[:success]
-      render "create"
-    else
-      redirect_to not_found_path
-    end
-  end
-
   def update
     if @api_key.enabled
       render "revoke"
-    else
+    elsif deletion_enabled?
       render "delete"
+    else
+      raise NotImplementedError, "API key deletion is not implemented"
```

Comment:
Why is it not enabled? There's a delete method defined

Reply:
Not mounted in production basically

Reply:
It should actually be impossible to get here so I added this defensive check